src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -301,64 +298,74 @@ private static void printOutput(Map<GKInstance, Set<Map<GKInstance, GKInstance>>
 
 	private static boolean hasNewInstanceEdit(List<GKInstance> currentRLEInstanceEdits, List<GKInstance> previousRLEInstanceEdits)
 	{
+		// start by assuming false and set to true when conditions require it.
+		boolean newInstanceEdit = false;
+		// I guess Justin assumed that if there are more InstanceEdits of the current version of an object, the it *must* be new.
+		// This is *probably* a safe assumption to make.
 		if (currentRLEInstanceEdits.size() != previousRLEInstanceEdits.size())
 		{
-			return true;
+			newInstanceEdit = true;
```

Comment:
This looks a bit dangerous.  If the size of the instance edits between the current and previous databases aren't equal, it is only a new instance edit if the current database has more than the previous database.  

This condition will also be true if the current database has less than the previous database which should log or trigger an error.

Reply:
I think Justin made an assumption that the number of InstanceEdits should always increase. _Are_ there any cases where the number of InstanceEdits for an entity _decreases_ over time?

I agree that it may be a foolish assumption, but I didn't design this.

Reply:
Not foolish - it makes sense this case wasn't added because this should never happen in proper handling of databases, but it has occurred before due to data issues.

Reply:
If there is precedent of it happening (yikes!), then I will make a note that ALL instance edits from both entities must be checked.